src/Indigo/Contracts/Holdings/LorentzBindings.hs

```diff
@@ -0,0 +1,86 @@
+-- SPDX-FileCopyrightText: 2020 TBD
+--
+-- SPDX-License-Identifier: LicenseRef-Proprietary
+-- | This module contains Lorentz entrypoints of ManagedLedger contract
+-- wrapped to produce Indigo code.
+module Indigo.Contracts.Holdings.LorentzBindings
+  ( approve
+  , burn
+  , ensureNotPaused
+  , getAllowance
+  , getBalance
+  , getTotalSupply
+  , mint
+  , setPause
+  , transfer
+  ) where
+
+import Indigo
+import qualified Lorentz as L
+
+import qualified Lorentz.Contracts.ManagedLedger.Impl as ML
+import qualified Lorentz.Contracts.Spec.ManagedLedgerInterface as ML
+
+
+import Indigo.Contracts.Holdings.Types
+
+ensureNotPaused :: HasStorage Storage => IndigoProcedure s
+ensureNotPaused = unaryOpFlat (storageVar @Storage) (ML.ensureNotPaused # L.drop)
+
+-- | Wrap lorentz entrypoint to produce Ingido code.
+-- Entrypoint produces new Storage and Ops, so the implicit Indigo variables
+-- for them are updated as well.
+lorentzEntrypointToIndigo
+  :: (TypeValue param, HasStorage Storage, HasSideEffects)
+  => L.Entrypoint param Storage
+  -> Var param -> IndigoM s ((Ops, Storage) & s) ()
+lorentzEntrypointToIndigo entrypointCode entrypointParam = do
+  entrypointRes <-
+    (fromLorentzFun2 $ L.framed entrypointCode) entrypointParam (storageVar @Storage)
+  let newStorage = snd entrypointRes
+      newOps = fst entrypointRes
+  -- Propagate storage changes
+  setVar (storageVar @Storage) newStorage
+  -- Add new chain operations
+  forEach newOps $ \op -> do
+    setVar operationsVar $ op .: operationsVar
```

Comment:
Is it not possible to base this on the Indigo implementation of `ManagedLedger`?
Or do you have other reasons not to do so?

Reply:
When I was implementing it previously, indigo version of ManagedLedger wasn't merged. Also, indigo version isn't polymorphic over storage type in contrast to Lorentz version

Reply:
Also, Lorentz implementation seems to be more efficient

Reply:
Ok, look like more than enough good reasons